core/mcp.go

```
@@ -0,0 +1,771 @@
+// ============================================================================
+
+// cleanup performs cleanup of all MCP resources.
+func (m *MCPManager) cleanup() error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	// Clean up STDIO processes
+	for _, client := range m.clientMap {
+		if client.StdioCommand != nil && client.StdioCommand.Process != nil {
+			m.logger.Info(fmt.Sprintf("%s Terminating STDIO process: %d", MCPLogPrefix, client.StdioCommand.Process.Pid))
+			client.StdioCommand.Process.Kill()
```

golangci-lint (1.64.8)

Error return value of `client.StdioCommand.Process.Kill` is not checked

(errcheck)